### ISSUE TITLE
Some APIs of JSON RPC returns empty Body with HTTP 200 status

### ISSUE THREAD

**Issue by y0ke** [rsksmart/rskj#408]
- Mac Bamboo v0.3 and v0.4 with [testnet config](https://github.com/rsksmart/artifacts/blob/master/rskj-ubuntu-installer/config/testnet.conf)
- Ubuntu 16 v0.3(rpm install) with [testnet config](https://github.com/rsksmart/artifacts/blob/master/rskj-ubuntu-installer/config/testnet.conf)

With the environment above, I encountered JSON RPC responses of empty HTTP200 status responses. some of APIs(ex. `eth_coinbase`,`eth_sendTransaction`) return a proper response body and some(ex. `eth_getTransactionByHash`, `eth_accounts`) return empty body.

I'm using curl and [Paw](https://paw.cloud/)

**Comment by herrerameri** [comment]
Hello @y0ke ! 
I'm trying the same using bash and my node connected to the testnet. 
But I'm getting expected results... 


- eth_accounts:
```
$ curl -s -X POST -H "Content-Type:application/json" -d '{"jsonrpc":"2.0","method":"eth_accounts", "params":{},"id":666}' http://localhost:4444
{"jsonrpc":"2.0","id":666,"result":[]}
```
(that is expected, because my `wallet` section in the config file is: `accounts = []`)
- eth_getTransactionByHash
I've searched a tx hash belonging to a block that my node has syncronized:
```
curl -s -X POST -H "Content-Type:application/json" -d '{"jsonrpc":"2.0","method":"eth_getTransactionByHash", "params":["0xff6dbd98ff8af4e3f42be9f00549cb91b03a22ef48829bc32b41741d311976fd"],"id":666}' http://localhost:4444
{"jsonrpc":"2.0","id":666,"result":{"hash":"0xff6dbd98ff8af4e3f42be9f00549cb91b03a22ef48829bc32b41741d311976fd","nonce":"0x3907","blockHash":"0xf66f312a20f95f57e6e6d12c76d21e0836e4eaf14bfd5bd21c3d1a54ac4068b1","blockNumber":"0x3908","transactionIndex":"0x0","from":"0x00","to":"0x0000000000000000000000000000000001000008","gas":"0x00","gasPrice":"0x00","value":"0x00","input":"0x00"}}
```

Can you try something similar with bash and tell me which is your result? 
Thanks!

**Comment by y0ke** [comment]
Hi @herrerameri 
Thanks for the reply!
I checked your snippet and got an expected response.
I compared my request history and your snippet and noticed that my requests were missing "id". 
It seems If a request doesn't have "id" parameter, response will be returned with empty body.
But anyways, Thanks It's solved.

This should be labeled as "enhance" though,  That'll be helpful if a node returns an error or expected response.

here is how geth and bitcoind handles a request with no "id" parameter
- geth
```
{"jsonrpc":"2.0","error":{"code":-32700,"message":"missing request id"}}
```
- bitcoind
expected response as  `{"id": null}`
